src/traces/streamtube/attributes.js

```diff
@@ -0,0 +1,141 @@
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var scatterAttrs = require('../scatter/attributes');
+var colorAttributes = require('../../components/colorscale/color_attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var scatterMarkerAttrs = scatterAttrs.marker;
+
+function makeProjectionAttr(axLetter) {
+    return {
+        show: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: false,
+            description: [
+                'Sets whether or not projections are shown along the',
+                axLetter, 'axis.'
+            ].join(' ')
+        },
+        opacity: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            description: 'Sets the projection color.'
+        },
+        scale: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 10,
+            dflt: 2 / 3,
+            description: [
+                'Sets the scale factor determining the size of the',
+                'projection marker points.'
+            ].join(' ')
+        }
+    };
+}
+
+module.exports = {
+    x: {
+        valType: 'data_array',
+        description: 'Sets the x coordinates.'
+    },
+    y: {
+        valType: 'data_array',
+        description: 'Sets the y coordinates.'
+    },
+    z: {
+        valType: 'data_array',
+        description: 'Sets the z coordinates.'
+    },
+    text: extendFlat({}, scatterAttrs.text, {
+        description: [
+            'Sets text elements associated with each (x,y,z) triplet.',
+            'If a single string, the same string appears over',
+            'all the data points.',
+            'If an array of string, the items are mapped in order to the',
+            'this trace\'s (x,y,z) coordinates.'
+        ].join(' ')
+    }),
+    mode: extendFlat({}, scatterAttrs.mode,  // shouldn't this be on-par with 2D?
+        {dflt: 'lines+markers'}),
+    projection: {
+        x: makeProjectionAttr('x'),
+        y: makeProjectionAttr('y'),
+        z: makeProjectionAttr('z')
+    },
+    sizingaxis: {
+        valType: 'enumerated',
+        role: 'info',
+        values: [0, 1, 2],
+        dflt: 0,
+        description: [
+            'Specifies the axis index in relation to which the `line.width` and `marker.size` values are determined. The',
+            'default value is `0`, which specifies the `x` axis, i.e. sizes will be determined as a multiple of one unit',
+            'on the `x` axis. `0`, `1`, `2` refer to `x`, `y`, `z`, respectively.'
+        ].join(' ')
+    },
+    connectgaps: scatterAttrs.connectgaps,
+    line: extendFlat({}, {
+        connectiondiameter: extendFlat({}, scatterMarkerAttrs.size, {
+            dflt: 1,
+            description: 'Sets the radius of the line connection. Either a number, or an array with as many elements as the number of points.'
```

Comment:
Yes. In this case, the light reflection can be more of a distraction than a positive. So with this type of use, it's good to avoid using a specular light component.